### PR TITLE
Enhance aks user guide for ssl certs in key vault scenarios

### PR DESCRIPTION
**Pull request by galiacheng**
This pr is to enhance aks user guide for ssl certs in key vault scenarios:

- Add parameter tables for **TLS/SSL configuration** -> **Use KeyStores stored in Azure Key Vault**
- Add application gateway trusted root certificate item in **Networking** -> **Application Gateway Ingress Controller** -> **Identify an Azure Key Vault**
- Update service principal command to keep it the same with latest AZ CLI requires.

### CODE REVIEW

**Comment by galiacheng** [comment]
@majguo thanks for evocating the update of service principal command, I have updated the documents, could you please review at your convenience?

File: documentation/staging/content/managing-domains/aks/_index.md [hunk 1]
```diff
@@ -178,24 +196,25 @@ You must select one of the following three options, each described in turn.
...
 
 **Generate a self-signed frontend certificate**
 
 | Field | Description |
 |-------|-------------|
 | Trusted root certificate(.cer, .cert) | A trusted root certificate is required to allow back-end instances in the application gateway. The root certificate is a Base-64 encoded X.509(.CER) format root certificate. |
-| Service Principal | A Base64 encoded JSON string of a service principal for the selected subscription. You can generate one with command `az ad sp create-for-rbac --role Contributor --sdk-auth | base64 -w0`. On macOS omit the `-w0`. For more information, see [Create a service principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal). |
+| Service Principal | A Base64 encoded JSON string of a service principal for the selected subscription. You can generate one with command `az ad sp create-for-rbac --role Contributor --sdk-auth --scopes /subscriptions/<AZURE_SUBSCRIPTION_ID> \| base64 -w0`. For more information, see [Create a service principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal). |
```
Comment (majguo) [inline documentation/staging/content/managing-domains/aks/_index.md:218]: > `\| base64`

`\` should be removed?

File: documentation/staging/content/managing-domains/aks/_index.md [hunk 2]
```diff
@@ -178,24 +196,25 @@ You must select one of the following three options, each described in turn.
...
+| The name of the secret in the specified Key Vault whose value is the password for the front-end TLS/SSL certificate | Enter the name of the Azure Key Vault secret that holds the value of the password for the application gateway front-end SSL certificate. |
+| The name of the secret in the specified Key Vault whose value is the trusted root certificate data | A trusted root certificate is required to allow back-end instances in the application gateway. Enter the name of the Azure Key Vault secret that holds the value of the application gateway trusted root certificate data. Follow [Store the TLS/SSL certificate in the Key Vault](#store-the-tlsssl-certificate-in-the-key-vault) to upload certificate to Azure Key Vault. |
+| Service Principal | A Base64 encoded JSON string of a service principal for the selected subscription. You can generate one with command `az ad sp create-for-rbac --role Contributor --sdk-auth --scopes /subscriptions/<AZURE_SUBSCRIPTION_ID> \| base64 -w0`. For more information, see [Create a service principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal). |
```
Comment (majguo) [inline documentation/staging/content/managing-domains/aks/_index.md:211]: > `\| base64 -w0`

`\` should be removed?

File: documentation/staging/content/managing-domains/aks/_index.md [hunk 3]
```diff
@@ -178,24 +196,25 @@ You must select one of the following three options, each described in turn.
 | Password | The password for the certificate |
 | Confirm password | Re-enter the value of the preceding field. |
 | Trusted root certificate(.cer, .cert) | A trusted root certificate is required to allow back-end instances in the application gateway. The root certificate is a Base-64 encoded X.509(.CER) format root certificate. |
-| Service Principal | A Base64 encoded JSON string of a service principal for the selected subscription. You can generate one with command `az ad sp create-for-rbac --role Contributor --sdk-auth | base64 -w0`. On macOS omit the `-w0`. For more information, see [Create a service principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal). |
+| Service Principal | A Base64 encoded JSON string of a service principal for the selected subscription. You can generate one with command `az ad sp create-for-rbac --role Contributor --sdk-auth --scopes /subscriptions/<AZURE_SUBSCRIPTION_ID> \| base64 -w0`. For more information, see [Create a service principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal). |
```
Comment (majguo) [inline documentation/staging/content/managing-domains/aks/_index.md:200]: > `\| base64 -w0`

`\` should be removed?

Besides, why "On macOS omit the `-w0`" is removed?

**Comment by majguo** [comment]
Note: The above comments are also applied to `_index.md` for `3.3` and `3.4`.

**Review by majguo — CHANGES_REQUESTED** [review]

File: documentation/staging/content/managing-domains/aks/_index.md (hunk 1 above)
Comment (galiacheng) [reply documentation/staging/content/managing-domains/aks/_index.md:218]: No, if remove "\", the command will show up as text, not a command.

File: documentation/staging/content/managing-domains/aks/_index.md (hunk 3 above)
Comment (galiacheng) [reply documentation/staging/content/managing-domains/aks/_index.md:200]: On macOS, no `-w0` required to generate one-line base64 string, which was tested by Ed, the offer UI definition also has this comment.

File: documentation/staging/content/managing-domains/aks/_index.md (hunk 1 above)
Comment (galiacheng) [reply documentation/staging/content/managing-domains/aks/_index.md:218]: Now, the command will show up as the picture shows:
image

File: documentation/staging/content/managing-domains/aks/_index.md (hunk 2 above)
Comment (galiacheng) [reply documentation/staging/content/managing-domains/aks/_index.md:211]: See [above comment.](https://github.com/oracle/weblogic-kubernetes-operator/pull/3174#discussion_r898653096)

File: documentation/staging/content/managing-domains/aks/_index.md (hunk 1 above)
Comment (majguo) [reply documentation/staging/content/managing-domains/aks/_index.md:218]: Thanks for letting me know.

File: documentation/staging/content/managing-domains/aks/_index.md (hunk 3 above)
Comment (majguo) [reply documentation/staging/content/managing-domains/aks/_index.md:200]: Looks good.

**Review by majguo — APPROVED** [review]

File: documentation/3.3/content/userguide/aks/_index.md [hunk 4]
```diff
@@ -133,7 +134,24 @@ If you want to upload existing keystores, select **Upload existing KeyStores** f
...
+
+If you want to use keystores that are stored in Azure Key Vault, select **Use KeyStores stored in Azure Key Vault** for the option **How would you like to provide required configuration**, and enter the values for the fields listed in the following table.
+
+##### Use KeyStores stored in Azure Key Vault
+
+| Field | Description |
+|-------|-------------|
+| Resource group name in current subscription containing the Key Vault | Enter the name of the Resource Group containing the Key Vault that stores the SSL certificate and the data required for WebLogic SSL termination. |
+| Name of the Azure Key Vault containing secrets for the TLS/SSL certificate | Enter the name of the Azure Key Vault that stores the SSL certificate and the data required for WebLogic SSL termination. |
+| The name of the secret in the specified Key Vault whose value is the Identity KeyStore Data | Enter the name of the Azure Key Vault secret that holds the value of the identity keystore data. Follow [Store the TLS/SSL certificate in the Key Vault](#store-the-tlsssl-certificate-in-the-key-vault) to upload certificate to Azure Key Vault. |
```
Comment (rosemarymarano) [inline documentation/3.3/content/userguide/aks/_index.md:147]: upload certificate -> upload the certificate (please correct all instances; I found 13.)

File: documentation/3.3/content/userguide/aks/_index.md [hunk 5]
```diff
@@ -270,3 +289,27 @@ In the **Review + create blade**, review the details you provided for deploying
 If you want to use this template to automate the deployment, download it by selecting **Download a template for automation**.
 
 Click **Create** to create this offer. This process may take 30 to 60 minutes.
+
+
+#### Useful resources
```
Comment (rosemarymarano) [inline documentation/3.3/content/userguide/aks/_index.md:294]: You need some text between these two headings. I can be something simple like, "Review the following useful resources."

**Review by rosemarymarano — APPROVED** [review]
very minor edits

**Comment by rjeberhard** [comment]
@rosemarymarano, I see that you approved, but there are still some unresolved comments. @galiacheng, given that I'll merge this PR as soon as you tell me that you've resolved the comments that you want to before being done with this PR.

File: documentation/3.3/content/userguide/aks/_index.md (hunk 4 above)
Comment (galiacheng) [reply documentation/3.3/content/userguide/aks/_index.md]: See https://github.com/oracle/weblogic-kubernetes-operator/pull/3174/commits/83051c4dc5feb07e74d5e8ac0e3ddcd05d60b3ec

File: documentation/3.3/content/userguide/aks/_index.md (hunk 5 above)
Comment (galiacheng) [reply documentation/3.3/content/userguide/aks/_index.md:293]: See https://github.com/oracle/weblogic-kubernetes-operator/pull/3174/commits/83051c4dc5feb07e74d5e8ac0e3ddcd05d60b3ec

**Review by edburns — COMMENTED** [review]
LGTM

**Pull request merged**